packages/gafl-webapp-service/src/pages/summary/contact-summary/__tests__/route.spec.js

```diff
@@ -33,349 +36,274 @@ const route = require('../route.js').default
-  premises: '14 Howecroft Court',
-  street: 'Eastmead Lane',
-  town: 'Bristol',
-  postcode: 'BS9 1HJ',
-  email: 'fester@tester.com',
-  mobilePhone: '01234567890'
-}
+const getMockCatalog = overrides => ({
+  change_licence_details_other: Symbol('Review or change the licence details other'),
+  change_licence_details_you: Symbol('Review or change the licence details'),
+  contact_summary_change: 'contact-summary-change',
+  contact_summary_email: 'contact-summary-email',
+  contact_summary_hidden_address: Symbol('contact-summary-hidden-address'),
+  contact_summary_hidden_licence_fulfilment: Symbol('contact-summary-hidden-licence-fulfilment'),
+  contact_summary_hidden_licence_confirmation: Symbol('contact-summary-hidden-licence-confirmation'),
+  contact_summary_hidden_contact: Symbol('contact-summary-hidden-contact'),
+  contact_summary_hidden_newsletter: Symbol('contact-summary-hidden-newsletter'),
+  contact_summary_license_default: 'contact-summary-license-default',
+  contact_summary_license_non_physical: 'contact-summary-license-non-physical',
+  contact_summary_license_physical: 'contact-summary-license-physical',
+  contact_summary_row_address: Symbol('contact-summary-row-address'),
+  contact_summary_row_licence: Symbol('contact-summary-row-licence'),
+  contact_summary_row_licence_conf: Symbol('contact-summary-row-licence-conf'),
+  contact_summary_row_contact: Symbol('contact-summary-row-contact'),
+  contact_summary_row_licence_details: Symbol('contact-summary-row-licence-details'),
+  contact_summary_row_newsletter: Symbol('contact-summary-row-newsletter'),
+  contact_summary_text_sngl: 'contact-summary-text-sngl',
+  contact_summary_text_plrl: 'contact-summary-text-plrl',
+  contact_summary_title: Symbol('contact-summary-title'),
+  no: 'negative, Ghost Rider',
+  yes: 'aye',
+  ...overrides
+})
+
+const getMockPermission = (licenseePermissions, permissions = {}) => ({
+  licenceLength: '12M',
+  isLicenceForYou: true,
+  ...permissions,
+  licensee: {
+    firstName: 'Fester',
+    lastName: 'Tester',
+    premises: '14 Howecroft Court',
+    street: 'Eastmead Lane',
+    town: 'Bristol',
+    postcode: 'BS9 1HJ',
+    email: 'fester@tester.com',
+    mobilePhone: '01234567890',
+    birthDate: '1996-01-01',
+    postalFulfilment: true,
+    ...licenseePermissions
+  }
+})
 
-const generateRequestMock = (permission, query, status = {}) => ({
+const getRequestMock = ({ permission = getMockPermission(), query, status, setCurrentPermission, catalog = getMockCatalog() } = {}) => ({
   cache: () => ({
     helpers: {
       status: {
         getCurrentPermission: async () => ({
+          fromSummary: false,
           [ADDRESS_ENTRY.page]: true,
           [ADDRESS_SELECT.page]: true,
           [CONTACT.page]: true,
           [LICENCE_FULFILMENT.page]: true,
           [LICENCE_CONFIRMATION_METHOD.page]: true,
           ...status
         }),
-        setCurrentPermission: () => {}
+        setCurrentPermission
```

Comment:
I think the parameter should be called `setStatusPermission`, even though this will mean a less succinct reference here: `setCurrentPermission: setStatusPermission`.

Also, in every test where you're not at all interested in `status.setCurrentPermission`, because you've not set a default, you have to specify a value for it. Set the default as `() => {}`, then you can remove the specification of `setCurrentPermission` in all tests not interested in it.